### PR TITLE
Add EcDSA verifier

### PR DESCRIPTION
**Pull request by Aconitin**
# Description of change
Implements a `JwsVerifier` capable of verifying EcDSA signatures with algorithms `ES256` and `ES256K`.

## Links to any relevant issues
None.

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested
See tests.rs.

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wulfraem — CHANGES_REQUESTED** [review]
Thank your for the PR, @Aconitin! ^^

I added some comments, it would be nice if you could go through them and update the parts or reply if you disagree or if the wording is a bit off or unclear. ^^;;

When you're done with the changes, it would be cool, if you could run cargo-fmt to align the code style. As some of the settings use nightly features, `cargo +nightly fmt` must be used. ^^

Also already added the license headers to the new files and added `identity_ecdsa_verifier` to the workspace config (which right at the moment should lead to errors in pipelines as dependencies would not be found :D).

File: identity_ecdsa_verifier/src/secp256k1.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        };
+
+        let verifying_key: VerifyingKey = VerifyingKey::from(public_key);
+
+        let signature: Signature =
+            Signature::try_from(input.decoded_signature.deref()).map_err(|err| {
+                SignatureVerificationError::new(SignatureVerificationErrorKind::InvalidSignature)
+                    .with_source(err)
+            })?;
+
```
[the remark this answers is not in the document]
Comment (Aconitin) [reply identity_ecdsa_verifier/src/secp256k1.rs:85]: fine, but signature needs to be mutable then. I'll change it :)

File: identity_ecdsa_verifier/src/tests.rs
[the remark this answers is not in the document]
Comment (Aconitin) [reply identity_ecdsa_verifier/src/tests.rs:1]: done

**Review by wulfraem — APPROVED** [review]
Looks good to me, thank you! ^^

To be able to merge, we need #1365 merged into main and then pulled into this PR go get checks back to green. So we'll wait with the merge until then.

**Review by UMR1352 — APPROVED** [review]

**Pull request merged**